@xingrz it looks like an ifdef needs to be put around the isr for it to be compiled regularly

```
/__w/zephyr/zephyr/drivers/gpio/gpio_dw.c:387:13: error: ‘gpio_dw_isr’ defined but not used [-Werror=unused-function]
```

https://github.com/zephyrproject-rtos/zephyqr/actions/runs/8379204533/job/22945720630#step:12:4413

--- a/drivers/gpio/gpio_dw.c
+++ b/drivers/gpio/gpio_dw.c
@@ -384,6 +384,7 @@ static inline int gpio_dw_manage_callback(const struct device *port,
 	return gpio_manage_callback(&context->callbacks, callback, set);
 }
 
+#if DT_ANY_INST_HAS_PROP_STATUS_OKAY(interrupts)
 static void gpio_dw_isr(const struct device *port)
 {
 	struct gpio_dw_runtime *context = port->data;
@@ -396,6 +397,7 @@ static void gpio_dw_isr(const struct device *port)
 
 	gpio_fire_callbacks(&context->callbacks, port, int_status);
 }
+#endif /* DT_ANY_INST_HAS_PROP_STATUS_OKAY(interrupts) */
 
 static const struct gpio_driver_api api_funcs = {
 	.pin_configure = gpio_dw_config,

